### DISCUSSION TITLE
resume training with different numbers of nodes.

### DISCUSSION THREAD

**Discussion by tangjiasheng** [deepspeedai/DeepSpeed#1895] [Ideas]
I think it would be better to provide this ability to resume training with different amount of nodes. e.g. from 128 cards to 64 cards.
Since training is not a stable progress, it often break down. If I train models on cloud, resources without guarantee is so general when resume training. So, combine all saved checkpoints and split them again is a good way to save time for not waiting resources.

**Comment by tjruwase** [comment]
@tangjiasheng, thanks for sharing this idea. In fact, we have an [effort ](https://github.com/bigscience-workshop/Megatron-DeepSpeed/pull/239) along this line. But it has been slow going due to bandwidth issues. However, expressed interests from our valued users like you will help us prioritize appropriately. Hope you can engage with the discussion. Thanks!
